### PR TITLE
driver: watchdog: npcx: support longer watchdog timeout

### PR DESCRIPTION
**Pull request by ChiHuaL**
In the current driver, the longest time of the watchdog timeout is ~8 seconds because the pre-scalar is fixed at 32 (WDCP=5).
This commit removes this limitation by dynamically calculating the pre-scalar according to the watchdog timeout setting from the API.

### CODE REVIEW

File: drivers/watchdog/Kconfig.npcx [hunk 1]
```diff
@@ -12,12 +12,11 @@ config WDT_NPCX
 	  processors.
 	  Say y if you wish to use watchdog on NPCX MCU.
 
-config WDT_NPCX_DELAY_CYCLES
-	int "Number of delay cycles before generating watchdog event/signal"
+config WDT_NPCX_WARNING_LEADING_TIME_MS
```
[the remark this answers is not in the document]
Comment (ChiHuaL) [reply drivers/watchdog/Kconfig.npcx:15]: done.

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -236,6 +236,11 @@ Drivers and Sensors
 
 * W1
 
+* Watchdog
+
+  * Added :kconfig:option:`WDT_NPCX_WARNING_LEADING_TIME_MS` to set the leading warning time
+    in the unit of milli-second. Removed no longer used :kconfig:option:`WDT_NPCX_DELAY_CYCLES`.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:242]:
```suggestion
  * Added :kconfig:option:`CONFIG_WDT_NPCX_WARNING_LEADING_TIME_MS` to set the leading warning time
    in milliseconds. Removed no longer used :kconfig:option:`CONFIG_WDT_NPCX_DELAY_CYCLES`.
```

File: doc/releases/migration-guide-3.7.rst [hunk 3]
```diff
@@ -221,6 +221,16 @@ Serial
 Timer
 =====
 
+Watchdog
+========
+
+* The ``nuvoton,npcx-watchdog`` driver has been changed to extend the max timeout period.
+  The time of one watchdog count bacomes various with the different pre-scalar setting.
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:228]: "bacomes various with the" 
Typo and weird grammar.

File: doc/releases/migration-guide-3.7.rst [hunk 4]
```diff
@@ -221,6 +221,16 @@ Serial
...
+  Removed :kconfig:option:`WDT_NPCX_DELAY_CYCLES` because it is no longer suitable to
+  set the leading warning time.
+  Add the kconfig:option:`WDT_NPCX_WARNING_LEADING_TIME_MS` to set the leading warning time
+  in the unit of milli-second instead.
+
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:233]:
```suggestion
  Instead, added the kconfig:option:`CONFIG_WDT_NPCX_WARNING_LEADING_TIME_MS` to set
  the leading warning time in milliseconds.

```

File: drivers/watchdog/Kconfig.npcx [hunk 5]
```diff
@@ -12,12 +12,11 @@ config WDT_NPCX
...
+	int "Number of milli-second before generating watchdog event/signal"
```
Comment (aescolar) [inline drivers/watchdog/Kconfig.npcx:16]:
```suggestion
	int "Milliseconds before generating watchdog event/signal"
```

File: doc/releases/release-notes-3.7.rst (hunk 2 above)
Comment (ChiHuaL) [reply doc/releases/release-notes-3.7.rst]: done.

File: doc/releases/migration-guide-3.7.rst (hunk 3 above)
Comment (ChiHuaL) [reply doc/releases/migration-guide-3.7.rst]: done

File: doc/releases/migration-guide-3.7.rst (hunk 4 above)
Comment (ChiHuaL) [reply doc/releases/migration-guide-3.7.rst]: done

File: drivers/watchdog/Kconfig.npcx (hunk 5 above)
Comment (ChiHuaL) [reply drivers/watchdog/Kconfig.npcx]: done

**Review by fabiobaltieri — DISMISSED** [review]

**Review by keith-zephyr — DISMISSED** [review]

**Review by MulinChao — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@ChiHuaL merge conflict on the migration guide, can you rebase please?

**Comment by ChiHuaL** [comment]
> @ChiHuaL merge conflict on the migration guide, can you rebase please?

Done.

**Review by keith-zephyr — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**